domain/src/main/java/no/difi/meldingsutveksling/core/EDUCoreFactory.java

```diff
@@ -127,6 +144,127 @@ public EDUCore create(Message message, String senderOrgNr) {
+        ofNullable(sm.getSakssekvensnummer()).map(BigInteger::toString).ifPresent(noarksakType::setSaSeknr);
+        noarksakType.setSaAnsvinit(sm.getSaksansvarlig());
+        noarksakType.setSaAdmkort(sm.getAdministrativEnhet());
+        noarksakType.setSaOfftittel(sm.getOffentligTittel());
+
+        if (!(sm.getBasisregistrering().get(0) instanceof Journalpost)) {
+            throw new MeldingsUtvekslingRuntimeException(String.format("Basisregistrering in Arkivmelding %s not instance of Journalpost", cr.getConversationId()));
+        }
+        JournpostType journpostType = of.createJournpostType();
+        Journalpost jp = (Journalpost)  sm.getBasisregistrering().get(0);
+        ofNullable(jp.getJournalaar()).map(BigInteger::toString).ifPresent(journpostType::setJpJaar);
+        ofNullable(jp.getJournalsekvensnummer()).map(BigInteger::toString).ifPresent(journpostType::setJpSeknr);
+        ofNullable(jp.getJournalpostnummer()).map(BigInteger::toString).ifPresent(journpostType::setJpJpostnr);
+        ofNullable(jp.getJournaldato()).map(XMLGregorianCalendar::toString).ifPresent(journpostType::setJpJdato);
+        ofNullable(jp.getJournalposttype()).map(Journalposttype::value).ifPresent(journpostType::setJpNdoktype);
+        ofNullable(jp.getDokumentetsDato()).map(XMLGregorianCalendar::toString).ifPresent(journpostType::setJpDokdato);
+        ofNullable(sm.getSkjerming()).map(Skjerming::getSkjermingshjemmel).ifPresent(journpostType::setJpUoff);
+
+
+        jp.getKorrespondansepart().forEach(k -> {
+            AvsmotType avsmotType = of.createAvsmotType();
+            avsmotType.setAmNavn(k.getKorrespondansepartNavn());
+            if ("avsender".equals(k.getKorrespondanseparttype().value().toLowerCase())) {
+                avsmotType.setAmIhtype("0");
+            }
+            if ("mottaker".equals(k.getKorrespondanseparttype().value().toLowerCase())) {
+                avsmotType.setAmIhtype("1");
+            }
+            avsmotType.setAmAdmkort(k.getAdministrativEnhet());
+            avsmotType.setAmSbhinit(k.getSaksbehandler());
+            if (!jp.getAvskrivning().isEmpty()) {
+                Avskrivning avs = jp.getAvskrivning().get(0);
+                ofNullable(avs.getAvskrivningsmaate()).map(Avskrivningsmaate::value).ifPresent(avsmotType::setAmAvskm);
+                ofNullable(avs.getAvskrivningsdato()).map(XMLGregorianCalendar::toString).ifPresent(avsmotType::setAmAvskdato);
+                avsmotType.setAmAvsavdok(avs.getReferanseAvskrivesAvJournalpost());
+            }
+
+            journpostType.getAvsmot().add(avsmotType);
+        });
+
+        jp.getDokumentbeskrivelseAndDokumentobjekt().stream()
+                .filter(d -> d instanceof Dokumentbeskrivelse)
+                .forEach(b -> {
+                    Dokumentbeskrivelse db = (Dokumentbeskrivelse) b;
+                    db.getDokumentobjekt().forEach(dobj -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this lambda expression number of lines from 23 to at most 20. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1188)